sycl/test/basic_tests/code_location.cpp

```diff
@@ -0,0 +1,37 @@
+// RUN: %clangxx -fsycl -DNDEBUG %s -o %t.out
+// RUN: %t.out
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %t.out
+
+#include <CL/sycl.hpp>
+#include <cassert>
+#include <iostream>
+
+int main() {
+  auto code_loc = sycl::detail::code_location::current();
+  const char *funcName = "main";
+#ifdef NDEBUG
+  if (code_loc.fileName() != nullptr)
+    return 1;
+  if (code_loc.functionName() != funcName)
+    return 1;
+  if (code_loc.lineNumber() != 11)
+    return 1;
+  if (code_loc.columnNumber() != 19)
+    return 1;
+#else
+  assert((code_loc.fileName() != nullptr));
+  std::string str = code_loc.fileName();
+  assert(((str.find("code_location.cpp") != std::string::npos) &&
```

Comment:
shouldn't we check that the corresponding builtin is available?

Reply:
Yes, we shouldn't, because clang always has a builtin, but user's host compilator may not.